src/requests/request.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Nectary;
+
+/**
+ * A request will validate based on a set of rules
+ *
+ * A request can also be used to take in options and
+ * pull in the data required to fulfill a request.
+ */
+abstract class Request {
+  public function validate( $error_callback ) {
+    $rules = $this->validation_rules();
+
+    foreach ( $rules as $name => $check ) {
+      if ( true !== $check ) {
+        if ( is_array( $error_callback ) ) {
+          return call_user_func_array( $error_callback, [ $check, $this ] );
+        }
+
+        return $error_callback( $check, $this );
```

Comment:
@rraub Should a request accumulate errors, or fail after the first error?

Since requests are automatically made, I'm thinking there should be two abstract classes, one with each type of functionality. But what should the default be?